### ISSUE TITLE
[Documentation]: request: tutorial on editing NWB files

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/pynwb#1773]
### What would you like changed or added to the documentation and why?

I'd like a tutorial that steps the user through the different use-cases of how to edit an NWB file.

1. changing the value of a dataset
2. changing the value of an attribute
3. overwriting a dataset
4. anything else

### Do you have any interest in helping write or edit the documentation?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this change was not already [requested](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

**Comment by mavaylon1** [comment]
Agreed. I'll take this on because I would also like this clearly demonstrated.

**Comment by mavaylon1** [comment]
@bendichter What about an expansion to the export tutorial?
@rly Thoughts?

**Comment by rly** [comment]
I agree - this is important. 

Because I was curious about attributes, I did some tests here a little bit ago: https://gist.github.com/rly/be7bee420b482b9ddcd084f57cc4115e

I think this could expand the export tutorial but should be structured like @bendichter mentions. A user comes in and wants to change X. They should be able to scan through a table of contents and get to a section that tells them exactly what they should do.
